### PR TITLE
Enabled golangci-lint

### PR DESCRIPTION
**Pull request by sayboras**
# Description

Related to #648 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #648 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

**Comment by youngbupark** [comment]
@sayboras LGTM. Thanks for your contribution! :) 

@yaron2 Looks like actor reminder and timer tests are flaky in macos. https://dev.azure.com/azure-octo/Dapr/_build/results?buildId=2587&view=logs&j=ed5cc819-a1c4-591f-e16b-8168661594f6&s=d654deb9-056d-50a2-1717-90c08683d50a

Rerunning build is zero-issue.
I'm not sure it is macos hosted pool issue or not.  Also we need to make pipeline public to help our contributors to see their build status and errors as soon as possible

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  concurrency: 4
+
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  deadline: 5m
+
+  # exit code when at least one issue was found, default is 1
+  issues-exit-code: 1
+
+  # include test files or not, default is true
+  tests: false
```
Comment (youngbupark) [inline .golangci.yml]: We would need to enable this in the next pr and fix lint in tests.

**Comment by yaron2** [comment]
> @sayboras LGTM. Thanks for your contribution! :)
> 
> @yaron2 Looks like actor reminder and timer tests are flaky in macos. https://dev.azure.com/azure-octo/Dapr/_build/results?buildId=2587&view=logs&j=ed5cc819-a1c4-591f-e16b-8168661594f6&s=d654deb9-056d-50a2-1717-90c08683d50a
> 
> Rerunning build is zero-issue.
> I'm not sure it is macos hosted pool issue or not. Also we need to make pipeline public to help our contributors to see their build status and errors as soon as possible

I've found a probable cause for this. Fixed with #651.

File: .golangci.yml (hunk 1 above)
Comment (sayboras) [reply .golangci.yml]: agreed, I will raise one issue for this once this PR is merged.

**Comment by sayboras** [comment]
@youngbupark please help to review again, I realised that the lint step was not enabled in azure pipeline, the same had been added.

@yaron2 for your review as well.

File: build-binary-template.yml [hunk 2]
```diff
@@ -31,6 +31,14 @@ jobs:
           shopt -s extglob
           mv !(gopath) '$(modulePath)'
           echo '##vso[task.prependpath]$(GOBIN)'
+          curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b '$(GOPATH)/bin' v1.21.0
```
Comment (sayboras) [inline build-binary-template.yml]: I think this command failed on windows, but I have no access to check the actual build log.

Comment (yaron2) [reply build-binary-template.yml]: We'll get build logs sorted out asap

**Comment by youngbupark** [comment]
@sayboras sorry for your inconvenience. We will try to make CI public on Monday. Please wait until Monday.

**Comment by youngbupark** [comment]
@sayboras golangci-lint is installed successfully, but there are still lint errors.

```
========================== Starting Command Output ===========================
"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "d:\a\_temp\334bfb20-e110-4fe2-a922-2173f4fecfd9.cmd""
golangci-lint.exe run
pkg\client\clientset\versioned\typed\components\v1alpha1\fake\fake_component.go:78         whitespace  unnecessary trailing newline
pkg\client\clientset\versioned\typed\configuration\v1alpha1\fake\fake_configuration.go:78  whitespace  unnecessary trailing newline
pkg\client\informers\externalversions\generic.go:63                                        whitespace  unnecessary trailing newline
pkg\client\clientset\versioned\fake\register.go:33:5                                       varcheck    `parameterCodec` is unused
pkg\version\version.go:1                                                                   gofmt       File is not `gofmt`-ed with `-s`
pkg\modes\modes.go:1                                                                       gofmt       File is not `gofmt`-ed with `-s`
pkg\config\modes\kubernetes_config.go:1                                                    gofmt       File is not `gofmt`-ed with `-s`
pkg\client\clientset\versioned\clientset.go:70:26                                          stylecheck  ST1005: error strings should not be capitalized
mingw32-make: *** [Makefile:250: lint] Error 1
##[error]Cmd.exe exited with code '2'.
Finishing: Run make lint

```

**Comment by sayboras** [comment]
@youngbupark thanks for your build log, seems like it's related to skip dir configuration, I am using unix like path. Will update for windows.

FYI. These dirs are containing auto generated files only, hence I decided to skip them for linting.

File: build-binary-template.yml (hunk 2 above)
Comment (sayboras) [reply build-binary-template.yml]: thanks 💯, i will try to wrap it up once having access to build log later :)

**Comment by youngbupark** [comment]
@sayboras we decide to migrate azure pipeline to github actions. I am working on the migration. ETA will be tomorrow. Meanwhile, I will copy the logs for you. 

```
========================== Starting Command Output ===========================
"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "d:\a\_temp\605d7552-92a7-416d-bba9-a2fb5259c044.cmd""
golangci-lint.exe run
pkg\apis\configuration\register.go:1  goimports  File is not `goimports`-ed with -local github.com/org/project
pkg\modes\modes.go:1                  goimports  File is not `goimports`-ed with -local github.com/org/project
pkg\apis\components\register.go:1     goimports  File is not `goimports`-ed with -local github.com/org/project
mingw32-make: *** [Makefile:250: lint] Error 1
##[error]Cmd.exe exited with code '2'.
Finishing: Run make lint
```

**Comment by youngbupark** [comment]
@sayboras I'm migrating Azure pipeline to GitHub actions, which is not fun. - https://github.com/youngbupark/dapr/runs/269215304

I will complete this migration by the end of tomorrow. Sorry for the inconvenience.

**Comment by youngbupark** [comment]
@sayboras  error logs
```
golangci-lint.exe run
pkg\apis\components\register.go:1        gofmt  File is not `gofmt`-ed with `-s`
pkg\modes\modes.go:1                     gofmt  File is not `gofmt`-ed with `-s`
pkg\config\modes\standalone_config.go:1  gofmt  File is not `gofmt`-ed with `-s`
mingw32-make: *** [Makefile:250: lint] Error 1
```

**Comment by sayboras** [comment]
> @sayboras error logs
> 
> ```
> golangci-lint.exe run
> pkg\apis\components\register.go:1        gofmt  File is not `gofmt`-ed with `-s`
> pkg\modes\modes.go:1                     gofmt  File is not `gofmt`-ed with `-s`
> pkg\config\modes\standalone_config.go:1  gofmt  File is not `gofmt`-ed with `-s`
> mingw32-make: *** [Makefile:250: lint] Error 1
> ```

I just give up on windows now, I might need to get a win machine `just` to fix this 😢

**Comment by youngbupark** [comment]
@sayboras I will try with my windows machine tomorrow after migration is done. It is not worth spending your time now.

**Comment by youngbupark** [comment]
This is the log from the last build

```
C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "d:\a\_temp\1359b333-3a7c-45b0-b8b8-f4c82f02c59a.cmd""
golangci-lint.exe run
pkg\apis\components\register.go:1        goimports  File is not `goimports`-ed with -local github.com/dapr/dapr/*
pkg\config\modes\standalone_config.go:1  goimports  File is not `goimports`-ed with -local github.com/dapr/dapr/*
pkg\modes\modes.go:1                     goimports  File is not `goimports`-ed with -local github.com/dapr/dapr/*
mingw32-make: *** [Makefile:250: lint] Error 1
##[error]Cmd.exe exited with code '2'.
```

**Comment by sayboras** [comment]
@youngbupark thanks a lot for your kind help. The build is passing now after disabled two linters which only failed in windows 😢

**Comment by youngbupark** [comment]
> @youngbupark thanks a lot for your kind help. The build is passing now after disabled two linters which only failed in windows 😢

Nice :) I will approve it. let's look at the two linters later.

**Comment by yaron2** [comment]
Kudos for not giving up @sayboras

**Pull request merged**